apis/fluentbit/v1alpha2/collector_types.go

```diff
@@ -0,0 +1,141 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/fluent/fluent-operator/pkg/utils"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CollectorSpec defines the desired state of FluentBit
+type CollectorSpec struct {
+	// Fluent Bit image.
+	Image string `json:"image,omitempty"`
+	// Fluent Bit Watcher command line arguments.
+	Args []string `json:"args,omitempty"`
+	// Fluent Bit image pull policy.
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// Fluent Bit image pull secret
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	// Storage for position db. You will use it if tail input is enabled.
+	PositionDB corev1.VolumeSource `json:"positionDB,omitempty"`
+	// Container log path
+	ContainerLogRealPath string `json:"containerLogRealPath,omitempty"`
+	// Compute Resources required by container.
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// NodeSelector
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Pod's scheduling constraints.
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// Tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// Fluentbitconfig object associated with this Fluentbit
+	FluentBitConfigName string `json:"fluentBitConfigName,omitempty"`
+	// The Secrets are mounted into /fluent-bit/secrets/<secret-name>.
+	Secrets []string `json:"secrets,omitempty"`
+	// RuntimeClassName represents the container runtime configuration.
+	RuntimeClassName string `json:"runtimeClassName,omitempty"`
+	// PriorityClassName represents the pod's priority class.
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+	// List of volumes that can be mounted by containers belonging to the pod.
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+	// Pod volumes to mount into the container's filesystem.
+	VolumesMounts []corev1.VolumeMount `json:"volumesMounts,omitempty"`
+	// Annotations to add to each Fluentbit pod.
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// SecurityContext holds pod-level security attributes and common container settings.
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
+	// Host networking is requested for this pod. Use the host's network namespace. If this option is set, the ports that will be used must be specified. Default to false.
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// Buffer definition
+	BufferVolume *BufferVolume `json:"buffer,omitempty"`
```

Comment:
Why need this? This volume is just created but not mounted.

Reply:
Mounted here
https://github.com/fluent/fluent-operator/pull/484/files#diff-ad79a92078fc8d5c14743ecf902fe8ddcf3ee9d0e79e2e0cc5fd39062f14811bR167

Reply:
What does `BufferVolume` do?